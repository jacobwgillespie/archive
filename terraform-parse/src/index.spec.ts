/* eslint-disable unicorn/no-abusive-eslint-disable */
/* eslint-disable */

import path from 'path'
import {parseModule} from '.'

it('should correctly parse HCL module', async () => {
  const metadata = await parseModule(path.join(__dirname, '../support/module'))
  expect(metadata).toMatchSnapshot()
})

it('should error when parsing a broken HCL module', async () => {
  const metadata = parseModule(path.join(__dirname, '../support/broken-module'))
  return expect(metadata).rejects.toMatchSnapshot()
})
