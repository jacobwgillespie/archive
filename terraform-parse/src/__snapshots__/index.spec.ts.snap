// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly parse HCL module 1`] = `
Object {
  "outputs": Array [],
  "variables": Array [
    Object {
      "description": "Description",
      "name": "a",
      "required": true,
      "type": "dynamic",
    },
    Object {
      "name": "b",
      "required": true,
      "type": "string",
    },
    Object {
      "name": "c",
      "required": false,
      "type": Array [
        "list",
        "string",
      ],
    },
    Object {
      "description": "Description",
      "name": "d",
      "required": true,
      "type": Array [
        "map",
        Array [
          "object",
          Object {
            "a": "string",
            "b": "number",
            "c": Array [
              "set",
              "string",
            ],
          },
        ],
      ],
    },
  ],
}
`;

exports[`should error when parsing a broken HCL module 1`] = `[Error: module/main.tf:2,10-14: Invalid type specification; The keyword "asdf" is not a valid type specification.]`;
